base/compiler/abstractinterpretation.jl

```diff
@@ -1144,7 +1144,10 @@ function abstract_eval_statement(interp::AbstractInterpreter, @nospecialize(e),
         end
         callinfo = abstract_call(interp, ea, argtypes, sv)
         sv.stmt_info[sv.currpc] = callinfo.info
-        t = callinfo.rt
+        rt = callinfo.rt
+        t = isa(rt, InterConditional) ?
+            transform_from_interconditional(rt, ea) :
```

Comment:
I think this place is too high level for this transformation. It should probably be inside abstract_call or at the site where we're pulling this out of the cache

Reply:
yeah I agree, I put this into `abstract_call`.